src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/steg/StegService.kt

```diff
@@ -0,0 +1,130 @@
+                hentBehandledeSteg(behandling, behandledeSteg).behandlingStegStatus = BehandlingStegStatus.KLAR
+                behandlingRepository.saveAndFlush(oppdaterBehandlingStatus(behandling, behandledeSteg))
+
+                hentStegInstans(behandledeSteg).gjenopptaSteg(behandlingId)
+            }
+            // AVBRUTT kan brukes kun for henleggelse
+            // TILBAKEFØRT steg blir oppdatert til KLAR når det forrige steget er behandlet
+            BehandlingStegStatus.AVBRUTT, BehandlingStegStatus.TILBAKEFØRT ->
+                throw Feil(
+                    "Kan ikke behandle behandling $behandlingId " +
+                        "med steg $behandledeSteg med status ${BehandlingStegStatus.AVBRUTT}"
+                )
+        }
+    }
+
+    private fun valider(behandling: Behandling, behandledeSteg: BehandlingSteg) {
+        // valider om steget kan behandles
+        if (!behandledeSteg.kanStegBehandles()) {
+            throw Feil("Steget ${behandledeSteg.name} kan ikke behandles for behandling ${behandling.id}")
+        }
+        // valider om steget samsvarer med opprettet årsak til behandling
+        behandledeSteg.gyldigForÅrsaker.singleOrNull { it == behandling.opprettetÅrsak }
+            ?: throw Feil(
+                "Steget ${behandledeSteg.name} er ikke gyldig for behandling ${behandling.id} " +
+                    "med opprettetÅrsak ${behandling.opprettetÅrsak}"
+            )
+
+        // valider om behandling har et steg med stegstatus KLAR som er forrige enn behandledeSteg
+        val stegKlarForBehandling = behandling.behandlingStegTilstand.singleOrNull {
+            it.behandlingSteg.sekvens < behandledeSteg.sekvens &&
+                it.behandlingStegStatus == BehandlingStegStatus.KLAR
+        }
+        if (stegKlarForBehandling != null) {
+            throw Feil(
+                "Behandling ${behandling.id} har allerede et steg " +
+                    "${stegKlarForBehandling.behandlingSteg.name}} som er klar for behandling. " +
+                    "Kan ikke behandle ${behandledeSteg.name}"
+            )
+        }
+
+        // valider om inlogget bruker har riktig rolle
+        // TODO tilgangssjekk
+    }
+
+    fun hentNesteSteg(behandling: Behandling, behandledeSteg: BehandlingSteg): BehandlingSteg {
+        val nesteGyldigeStadier = BehandlingSteg.values().filter {
+            it.sekvens > behandledeSteg.sekvens &&
+                behandling.opprettetÅrsak in it.gyldigForÅrsaker
+        }.sortedBy { it.sekvens }
+        return when (behandledeSteg) {
+            BehandlingSteg.BEHANDLING_AVSLUTTET -> throw Feil("Behandling ${behandling.id} er allerede avsluttet")
+            BehandlingSteg.BESLUTTE_VEDTAK -> hentNesteStegTypeBasertPåBehandlingsresultat(behandling.resultat)
+            else -> nesteGyldigeStadier.first()
+        }
+    }
+
+    private fun hentNesteStegTypeBasertPåBehandlingsresultat(resultat: Behandlingsresultat): BehandlingSteg {
```

Comment:
Kan man bruke denne metoden når som helst, eller gjelder det bare når nåværende steg er BESLUTTE_VEDTAK?

Reply:
det gjelder bare for beslutte vedtak steg.

Reply:
Da lurer jeg på om vi bør få det fram litt mer tydelig i metodenavnet. 
Jeg bare antok at det var BESLUTTE_VEDTAK siden jeg sjekket hvor den ble brukt 😛

Reply:
endret metodenavn til hentNesteStegEtterBeslutteVedtakBasertPåBehandlingsresultat